tests/deploy/e2e/test_deploy.py

```diff
@@ -1068,3 +1068,78 @@ async def test_skipped_for_dependency(resource_container, server, client, client
         deployment_result=DeploymentResult.DEPLOYED,
         blocked=BlockedStatus.NO,
     )
+
+
+async def test_redeploy_after_dependency_recovered(resource_container, server, client, clienthelper, environment, agent):
+    """
+    Asserts that a transiently skipped resource recovers when its dependencies succeed
+    """
+    version = await clienthelper.get_version()
+
+    # Disable auto deploys and repairs
+    Config.set("config", "agent-repair-interval", "0")
+    Config.set("config", "agent-deploy-interval", "0")
+
+    resource_container.Provider.set_fail("agent1", "key", 1)
+
+    rid1 = "test::Resource[agent1,key=key]"
+    rid2 = "test::Resource[agent1,key=key2]"
+
+    resources = [
+        {
+            "key": "key",
+            "value": "value",
+            "id": f"{rid1},v={version}",
+            "requires": [],
+            "purged": False,
+            "send_event": True,
+            "receive_events": False,
```

Comment:
This is about a special event regardless of event settings, so I'd set it to the strictest mode: disable all events. Then we can verify that even then, we get the recovery.
```suggestion
            "send_event": False,
            "receive_events": False,
```